Crash on Android 14 Beta (Pixel 7 Pro)

Hi! I am currently unable to perform any actions (such as starting a recording, turning on GPS) in the OpenTracks app on my Pixel 7 Pro. All permissions requested (Location, Nearby Devices, Notifications) have been allowed, and I have tried reinstalling both the default and nightly versions of the app.

Any advice would be appreciated!

### App information
* ID: de.dennisguse.opentracks.nightly
* Version: 5318 v4.7.0-4-g10e479ceb

### Device information
* Brand: google
* Device: cheetah
* Model: Pixel 7 Pro
* Id: UPB4.230623.007
* Product: cheetah_beta

### Firmware
* SDK: 34
* Release: 14
* Incremental: 10531303

### Cause of error
```java
Exception in thread "main": java.lang.SecurityException: Starting FGS with type connectedDevice callerApp=ProcessRecord{c57f105 26577:de.dennisguse.opentracks.nightly/u0a262} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_CONNECTED_DEVICE] any of the permissions allOf=false [android.permission.BLUETOOTH_ADVERTISE, android.permission.BLUETOOTH_CONNECT, android.permission.BLUETOOTH_SCAN, android.permission.CHANGE_NETWORK_STATE, android.permission.CHANGE_WIFI_STATE, android.permission.CHANGE_WIFI_MULTICAST_STATE, android.permission.NFC, android.permission.TRANSMIT_IR, android.permission.UWB_RANGING, USB Device, USB Accessory] 
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
	at android.os.Parcel.createException(Parcel.java:3041)
	at android.os.Parcel.readException(Parcel.java:3024)
	at android.os.Parcel.readException(Parcel.java:2966)
	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6761)
	at android.app.Service.startForeground(Service.java:775)
	at de.dennisguse.opentracks.services.TrackRecordingService.startSensors(TrackRecordingService.java:208)
	at de.dennisguse.opentracks.services.TrackRecordingService.startRecording(TrackRecordingService.java:194)
	at de.dennisguse.opentracks.services.TrackRecordingService.startNewTrack(TrackRecordingService.java:175)
	at de.dennisguse.opentracks.TrackListActivity.lambda$onCreate$6(TrackListActivity.java:255)
	at de.dennisguse.opentracks.TrackListActivity.$r8$lambda$sfL1wLWtkeS4x8K11Too1HQMWNU(Unknown Source:0)
	at de.dennisguse.opentracks.TrackListActivity$$ExternalSyntheticLambda11.onConnected(Unknown Source:2)
	at de.dennisguse.opentracks.services.TrackRecordingServiceConnection.setTrackRecordingService(TrackRecordingServiceConnection.java:147)
	at de.dennisguse.opentracks.services.TrackRecordingServiceConnection.onServiceConnected(TrackRecordingServiceConnection.java:162)
	at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:2198)
	at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:2231)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8177)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2611)
	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2322)
	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1679)
	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13265)
	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3385)

```

--- a/src/main/AndroidManifest.xml
+++ b/src/main/AndroidManifest.xml
@@ -41,6 +41,7 @@ limitations under the License.
         android:required="false" />
 
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_CONNECTED_DEVICE" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
     <uses-permission android:name="android.permission.VIBRATE" />

--- a/src/main/java/de/dennisguse/opentracks/services/TrackRecordingService.java
+++ b/src/main/java/de/dennisguse/opentracks/services/TrackRecordingService.java
@@ -18,6 +18,8 @@ package de.dennisguse.opentracks.services;
 
 import android.app.Service;
 import android.content.Intent;
+import android.content.pm.ServiceInfo;
+import android.os.Build;
 import android.os.Handler;
 import android.os.Looper;
 import android.os.PowerManager.WakeLock;
@@ -205,7 +207,11 @@ public class TrackRecordingService extends Service implements TrackPointCreator.
     private void startSensors() {
         wakeLock = SystemUtils.acquireWakeLock(this, wakeLock);
         trackPointCreator.start(this, handler);
-        startForeground(TrackRecordingServiceNotificationManager.NOTIFICATION_ID, notificationManager.setGPSonlyStarted(this));
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            startForeground(TrackRecordingServiceNotificationManager.NOTIFICATION_ID, notificationManager.setGPSonlyStarted(this), ServiceInfo.FOREGROUND_SERVICE_TYPE_CONNECTED_DEVICE);
+        } else {
+            startForeground(TrackRecordingServiceNotificationManager.NOTIFICATION_ID, notificationManager.setGPSonlyStarted(this));
+        }
     }
 
     public void endCurrentTrack() {

--- a/src/main/java/de/dennisguse/opentracks/services/handlers/AltitudeCorrectionManager.java
+++ b/src/main/java/de/dennisguse/opentracks/services/handlers/AltitudeCorrectionManager.java
@@ -25,11 +25,11 @@ public class AltitudeCorrectionManager {
 
     public AltitudeCorrectionManager() {
         this.altitudeConverterFallback = new EGM2008Internal();
-        this.altitudeConverter = Build.VERSION.SDK_INT >= 34 ? new AltitudeConverter() : null;
+        this.altitudeConverter = Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE ? new AltitudeConverter() : null;
     }
 
     public void correctAltitude(Context context, TrackPoint trackPoint) {
-        if (Build.VERSION.SDK_INT >= 34 && altitudeConverter != null) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE && altitudeConverter != null) {
             try {
                 Location loc = trackPoint.getLocation();
                 altitudeConverter.addMslAltitudeToLocation(context, loc);

